File: Alternativas/ej6.py
# -*- coding: utf-8 -*-

#Programa que lea una cadena por teclado y compruebe si es una letra mayúscula.

#1.almacenar letra (str)
#2.volver a almacenar letra en ora variable (str)
#3.transformar una de las letras a mayuscula
#4.comparar ambas variables
#5.si son iguales la letra es mayuscula
#6.si por el contrarios son distintos la letra sera minuscula

print("---Descubrir si una letra es mayúscula---")
letra1 = input("Introducir letra(mayúscula o miniscula)> ")
if(len(letra1)==1):
    if(letra1.isdigit()):
        print("El caracter es un número")
    else:
        letra2 = letra1
        if((letra2.upper())==letra1):
            print(f"{letra1} es mayúscula")
        else:
            print(f"{letra1} es minúscula")

File: Java/Alternativas/ej6.java
import java.util.Scanner; //Importar la clase Scanner 

/**
 * Programa que lea una cadena por teclado y compruebe si es una letra mayúscula.
 * @author Alberto Miguel Martinez Jimenez
 */
/**
 * Algoritmo:
 * Almacenar letra
 * Duplicar variable letra
 * transformar copia de letra a mayúscula
 * comparamos ambas variables
 * si son iguales letra es mayúscula
 * si no es minúscula
 */

public class ej6 {

	public static void main(String[] args) {
		char letra;
		String entradaLetra;
		char letraCopia;
		Scanner scanner = new Scanner(System.in);//Intanciar el objeto Scanner
		
		System.out.println("---Comprobar si una letra es mayúscula o minúscula---");
		System.out.println("Introduce una letra> ");
		entradaLetra = scanner.next();
		letra = entradaLetra.charAt(0);
		letraCopia = Character.toUpperCase(entradaLetra.charAt(0));
		if(letra==letraCopia) {
			System.out.println("La letra introducida es mayúscula.");
		}else{
			System.out.println("La letra introducida es minúscula.");
		}
		
		
	}

}
